src/Features/LanguageServer/Protocol/Handler/Hover/HoverHandler.cs

```diff
@@ -47,17 +51,42 @@ public HoverHandler()
                 return null;
             }
 
-            var text = await document.GetTextAsync(cancellationToken).ConfigureAwait(false);
+            var supportsVSExtensions = context.ClientCapabilities.HasVisualStudioLspCapability() == true;
 
-            // TODO - Switch to markup content once it supports classifications.
-            // https://devdiv.visualstudio.com/DevDiv/_workitems/edit/918138
-            return new VSHover
+            var hover = await GetHoverAsync(info, document, supportsVSExtensions, cancellationToken).ConfigureAwait(false);
+            return hover;
+
+            static async Task<Hover> GetHoverAsync(QuickInfoItem info, Document document, bool supportsVSExtensions, CancellationToken cancellationToken)
             {
-                Range = ProtocolConversions.TextSpanToRange(info.Span, text),
-                Contents = new SumType<SumType<string, MarkedString>, SumType<string, MarkedString>[], MarkupContent>(string.Empty),
-                // Build the classified text without navigation actions - they are not serializable.
-                RawContent = await IntellisenseQuickInfoBuilder.BuildContentWithoutNavigationActionsAsync(info, document, cancellationToken).ConfigureAwait(false)
-            };
+                var text = await document.GetTextAsync(cancellationToken).ConfigureAwait(false);
+                if (supportsVSExtensions)
+                {
+                    return new VSHover
+                    {
+                        Range = ProtocolConversions.TextSpanToRange(info.Span, text),
+                        Contents = new SumType<SumType<string, MarkedString>, SumType<string, MarkedString>[], MarkupContent>(string.Empty),
+                        // Build the classified text without navigation actions - they are not serializable.
+                        // TODO - Switch to markup content once it supports classifications.
+                        // https://devdiv.visualstudio.com/DevDiv/_workitems/edit/918138
+                        RawContent = await IntellisenseQuickInfoBuilder.BuildContentWithoutNavigationActionsAsync(info, document, cancellationToken).ConfigureAwait(false)
+                    };
+                }
+                else
```

Comment:
Technically you should be inspecting the markdown capability here:
![image](https://user-images.githubusercontent.com/2008729/117555199-ece07e00-b011-11eb-8eee-b91fe4404a02.png)

If there's no markdown you'd want to return just the string content, if there is markdown you'd want to inspect the markdown parser name to ensure it's `marked` otherwise a lot of your assumptions on how you built the markdown may not be accurate. Granted, given the platforms that you're supporting this could just be all overly cautious

Reply:
So I've added a per feature check to ensure the feature allows markdown.  However, the general capability to determine markdown spec is not present in the protocol library.  I'll move forward on this PR without it and followup with a PR to the LSP protocol library to add a definition for it